x/cash/staticfee.go

```diff
@@ -133,10 +133,13 @@ func (d FeeDecorator) extractFee(ctx weave.Context, tx weave.Tx, store weave.KVS
 	}
 
 	cmp := gconf.Coin(store, GconfMinimalFee)
-	// minimum has no currency -> accept everything
+	if cmp.IsZero() {
+		return finfo, nil
+	}
 	if cmp.Ticker == "" {
-		cmp.Ticker = fee.Ticker
+		return nil, errors.Wrap(coin.ErrInvalidCurrency, "no ticker")
```

Comment:
This is fine, but isn't this also covered by the `!fee.SameType()` line below?

Reply:
True. Look like `Coin.IsGTE` is calling `SameType` as well, which means it would cover all three cases. I think I prefer verbosity here. Having those 3 rules tested explicitly made it easier for me to understand what is expected from the data in order to be valid.

I will merge it as it is right now. If you think this should be changed then I can create another pull request.